### PR TITLE
Include Kimi style muon learning rate scaling

### PR DESCRIPTION
**Pull request by WhenWen**
As our scaling law in the speedrun in Marin is more pessimistic than the Kimi's version, we want to sanity check whether it is because we are using different per-module learning rate scaling.

### CODE REVIEW

File: src/levanter/optim/muon.py [hunk 1]
```diff
@@ -133,7 +136,10 @@ def transform_linear_layer(layer: haliax.nn.Linear):
             array = layer.weight.array
             updated_weight_array = zeropower_via_newtonschulz5(array, steps=steps, eps=muon_eps)
 
-            scale = jnp.sqrt(jnp.maximum(1, updated_weight_array.shape[0] / updated_weight_array.shape[1]))
+            if not use_kimi_scaling:
+                scale = jnp.sqrt(jnp.maximum(1, updated_weight_array.shape[0] / updated_weight_array.shape[1]))
```
Comment (Helw150) [inline src/levanter/optim/muon.py]:
```suggestion
                scale = jnp.sqrt(jnp.maximum(1, updated_weight_array.shape[0] / updated_weight_array.shape[1])) # sqrt(Out/In)
```
Quick comment to enhance readability.

Aside, but in theory it would be nice if we could use named axes here to make sure this always corresponds to Out / In. 

I double checked that for all the MLPs, we have out first=True so this should be correct, but this feels like the type of code that NamedAxes are supposed to make typesafe.

Comment (dlwh) [reply src/levanter/optim/muon.py]: realistically optimizer people want to do their matrix stuff so we should support the flattening thing, but I agree we should double check on in/out. here it's probably easiest to just check if out_first=True

Comment (WhenWen) [reply src/levanter/optim/muon.py]: Will do
